File: taller-5-python-ronalcastro97/zooAnimales/pez.py
from zooAnimales.animal import Animal

class Pez(Animal):
    _listado=[]
    salmones=0
    bacalaos=0
    def __init__(self, nombre, edad, habitat, genero,colorEscamas,cantidadAletas):
        super().__init__(nombre, edad, habitat, genero)
        self._colorEscamas=colorEscamas
        self._cantidadAletas=cantidadAletas
        Pez._listado.append(self)

    def getColorEscamas(self):
        return self._colorEscamas
    
    def setColorEscamas(self,colorEscamas):
        self._colorEscamas=colorEscamas

    def getCantidadAletas(self):
        return self._cantidadAletas
    
    def setCantidadAletas(self,cantidadAletas):
        self._cantidadAletas=cantidadAletas
    
    @classmethod
    def crearSalmon(cls,nombre,edad,genero):
        cls.salmones+=1
        return cls(nombre,edad,"oceano",genero,"rojo",6)
    
    @classmethod
    def crearBacalao(cls,nombre,edad,genero):
        cls.bacalaos+=1
        return cls(nombre,edad,"oceano",genero,"gris",6)
    
    @staticmethod
    def cantidadPeces():
        return len(Pez._listado)

File: taller-5-java-ronalcastro97/src/main/java/zooAnimales/Pez.java
package zooAnimales;
import gestion.*;

import java.util.ArrayList;

public class Pez extends Animal{
	private static ArrayList<Pez>listado=new ArrayList<>();
	public static int salmones=0;
	public static int  bacalaos=0;
	private String colorEscamas;
	private int cantidadAletas;
	public Pez() {
		super();
        Pez.listado.add(this);

	}
	public Pez(String nombre, int edad, String habitat, String genero,String colorEscamas, int cantidadAletas) {
		super(nombre, edad, habitat, genero);
		this.colorEscamas=colorEscamas;
		this.cantidadAletas=cantidadAletas;
		Pez.listado.add(this);
	}
	public static ArrayList<Pez> getListado() {
		return listado;
	}
	public static void setListado(ArrayList<Pez> listado) {
		Pez.listado = listado;
	}
	public static int getSalmones() {
		return salmones;
	}
	public static void setSalmones(int salmones) {
		Pez.salmones = salmones;
	}
	public static int getBacalaos() {
		return bacalaos;
	}
	public static void setBacalaos(int bacalaos) {
		Pez.bacalaos = bacalaos;
	}
	public String getColorEscamas() {
		return colorEscamas;
	}
	public void setColorEscamas(String colorEscamas) {
		this.colorEscamas = colorEscamas;
	}
	public int getCantidadAletas() {
		return cantidadAletas;
	}
	public void setCantidadAletas(int cantidadAletas) {
		this.cantidadAletas = cantidadAletas;
	}
	
	public String movimiento() {
		return "nadar";	
}
	public static Pez crearSalmon(String nombre,int edad,String genero) {
		Pez salmon=new Pez(nombre,edad,"oceano",genero,"rojo",6);
		Pez.salmones++;
		return salmon;
	}
	public static Pez crearBacalao(String nombre,int edad,String genero) { 
		Pez bacalao=new Pez(nombre,edad,"oceano",genero,"gris",6); 
		Pez.bacalaos++;
		return bacalao;
	}
	public static int cantidadPeces() {
		return Pez.listado.size();
	}
}
